pinot-broker/src/test/java/org/apache/pinot/broker/routing/segmentpruner/SegmentPrunerTest.java

```diff
@@ -241,17 +242,17 @@ public void testPartitionAwareSegmentPruner(QueryCompiler compiler) {
         new PartitionSegmentPruner(OFFLINE_TABLE_NAME, PARTITION_COLUMN, _propertyStore);
     Set<String> onlineSegments = new HashSet<>();
     segmentPruner.init(externalView, idealState, onlineSegments);
-    assertEquals(segmentPruner.prune(brokerRequest1, Collections.emptySet()), Collections.emptySet());
-    assertEquals(segmentPruner.prune(brokerRequest2, Collections.emptySet()), Collections.emptySet());
-    assertEquals(segmentPruner.prune(brokerRequest3, Collections.emptySet()), Collections.emptySet());
+    assertEquals(segmentPruner.prune(brokerRequest1, new SelectedSegments(Collections.emptySet(), true)).getSegments(), Collections.emptySet());
```

Comment:
Lots of changes are due to the signature change.
We can avoid these changes by adding a "VisibleForTest" package private wrapper to segmentPruner.
For now let's keep the changes here